service/controller/v3/resource/vpngateway/update.go

```diff
@@ -0,0 +1,172 @@
+//     VirtualNetworkGatewayConnectionPropertiesFormat.VirtualNetworkGateway2.ID
+//     VirtualNetworkGatewayConnectionPropertiesFormat.ConnectionType
+//     VirtualNetworkGatewayConnectionPropertiesFormat.ConnectionStatus
+//
+func needUpdate(current, desired network.VirtualNetworkGatewayConnection) bool {
+	if desired.Name == nil ||
+		desired.VirtualNetworkGatewayConnectionPropertiesFormat == nil ||
+		desired.VirtualNetworkGatewayConnectionPropertiesFormat.VirtualNetworkGateway1 == nil ||
+		desired.VirtualNetworkGatewayConnectionPropertiesFormat.VirtualNetworkGateway1.ID == nil ||
+		desired.VirtualNetworkGatewayConnectionPropertiesFormat.VirtualNetworkGateway2 == nil ||
+		desired.VirtualNetworkGatewayConnectionPropertiesFormat.VirtualNetworkGateway2.ID == nil {
+		return false
+	}
+
+	if current.Name == nil ||
+		*current.Name != *desired.Name {
+		return true
+	}
+
+	if current.VirtualNetworkGatewayConnectionPropertiesFormat == nil {
+		return true
+	}
+
+	if current.VirtualNetworkGatewayConnectionPropertiesFormat.ConnectionType !=
+		desired.VirtualNetworkGatewayConnectionPropertiesFormat.ConnectionType {
+		return true
+	}
+
+	if current.VirtualNetworkGatewayConnectionPropertiesFormat.VirtualNetworkGateway1 == nil ||
+		current.VirtualNetworkGatewayConnectionPropertiesFormat.VirtualNetworkGateway1.ID == nil ||
+		*current.VirtualNetworkGatewayConnectionPropertiesFormat.VirtualNetworkGateway1.ID !=
+			*desired.VirtualNetworkGatewayConnectionPropertiesFormat.VirtualNetworkGateway1.ID {
+		return true
+	}
+
+	if current.VirtualNetworkGatewayConnectionPropertiesFormat.VirtualNetworkGateway2 == nil ||
+		current.VirtualNetworkGatewayConnectionPropertiesFormat.VirtualNetworkGateway2.ID == nil ||
+		*current.VirtualNetworkGatewayConnectionPropertiesFormat.VirtualNetworkGateway2.ID !=
+			*desired.VirtualNetworkGatewayConnectionPropertiesFormat.VirtualNetworkGateway2.ID {
+		return true
+	}
+
+	if current.VirtualNetworkGatewayConnectionPropertiesFormat.ConnectionStatus == network.VirtualNetworkGatewayConnectionStatusNotConnected {
+		return true
+	}
+
+	return false
+}
+
+// ApplyUpdateChange perform the host cluster vpn gateway connections update against azure.
+func (r *Resource) ApplyUpdateChange(ctx context.Context, azureConfig, change interface{}) error {
+	a, err := key.ToCustomObject(azureConfig)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	c, err := toVPNGatewayConnections(change)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	err = r.applyUpdateChange(ctx, a, c)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	return nil
+}
+
+func (r *Resource) applyUpdateChange(ctx context.Context, azureConfig providerv1alpha1.AzureConfig, change connections) error {
+	r.logger.LogCtx(ctx, "level", "debug", "message", "ensure vpn gateway connections")
+
+	if change.isEmpty() {
+		r.logger.LogCtx(ctx, "level", "debug", "message", "ensure vpn gateway connections: already ensured")
```

Comment:
As I mentioned this already a couple of times this message format is not what we use everywhere. It got introduced in the azure operator at some point for whatever reason but it is not really helping us IMO. I would like to stick to the messages as we have them everywhere else.

Reply:
done

https://github.com/giantswarm/azure-operator/pull/278/commits/ddbf286226a3146b429bfe48ffcb7ea5c341e7cb